pandas/formats/format.py

```diff
@@ -1951,7 +1951,12 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    if levels._can_hold_na:
+                        values = levels.take(labels, fill_value=True)
```

Comment:
@jorisvandenbossche You are right. It doesn't matter if a level contains NaNs, but if it can contain NaNs.

Unfortunately the approach with the one-liner doesn't work. I've taken the same DataFrame as above and changed `take(labels_0, fill_value=True)` to `take(labels_0, fill_falue=False)`. The method `take` still raises the same exception:

    levels_0 = df.index.levels[0]
    labels_0 = df.index.labels[0]
    values_0 = levels_0.take(labels_0, fill_value=False); values_0

    ...
    ValueError: Unable to fill values because Int64Index cannot contain NA

If the level cannot contain NaNs as labels, only `take(_, fill_value=None)` works.